lib/amplitude_api.rb

```diff
@@ -152,16 +156,9 @@ def identify(*identifications)
     #
     # @return [ Typhoeus::Response ]
     def segmentation(event, start_time, end_time, **options)
-      Typhoeus.get SEGMENTATION_URI_STRING, userpwd: "#{api_key}:#{secret_key}", params: {
-        e: event.to_json,
-        m: options[:m],
-        start: start_time.strftime('%Y%m%d'),
-        end: end_time.strftime('%Y%m%d'),
-        i: options[:i],
-        s: (options[:s] || []).map(&:to_json),
-        g: options[:g],
-        limit: options[:limit]
-      }.delete_if { |_, value| value.nil? }
+      Typhoeus.get SEGMENTATION_URI_STRING, timeout: config.timeout, connecttimeout: config.connecttimeout,
+                                            userpwd: "#{api_key}:#{secret_key}",
+                                            params: segmentation_params(event, start_time, end_time, options)
```

Comment:
had to extract this because rubocop was complaining about a method having too many lines.

Reply:
Something that would (a) help with the Rubocop issue and (b) allow us to potentially switch away from Typhoeus more easily, should we ever need to, is creating a custom interface for making these HTTP requests. Instead of `Typhoeus.get` we'd have something like `InterfaceClass.get`, and that custom interface would know how to take in a partial options hash and fill it with default values, such as `config.timeout` and `config.connecttimeout`.

An incomplete example:

```ruby
class Requestor
  class << self
    def get(url, options)
      Typhoeus.get(url, with_defaults(options)
    end

    def with_defaults(options)
      {
        timeout: config.timeout,
        connect_timeout: config.connect_timeout
      }.merge(options)
    end
  end
end
```

This would also simplify the test suite, because instead of checking that every single call sets `timeout` and `connect_timeout`, you would only need to test 1) that `with_defaults` is being called and 2) that `with_defaults` actually works as expected. It would also allow us to add `**options` to practically every call so that you could even set these timeouts dynamically when invoking the method.

Reply:
I was considering this option myself, didn't want to do a huge change :)

Let me create a separate PR for that and we can rebase this on that one and see. It might take some time because all the tests are currently mocking typheous. I was thinking about using webmock for the same reasons you listed.

Reply:
What do you think about using Faraday? 
It's already does the job of providing a standard interface for adapters.

Typheous also has one for it https://github.com/typhoeus/typhoeus/blob/master/lib/typhoeus/adapters/faraday.rb

Reply:
I would imagine it was just a personal preference kind of decision when it was first introduced. Changing it should be acceptable, assuming that layer of code is refactored into a separate interface. Then it's possible to, similar to what Typhoeus and others do, have multiple interfaces for different implementations and then it's up to the developer to import the correct implementation before use.